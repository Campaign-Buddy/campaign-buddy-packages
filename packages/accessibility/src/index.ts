export * from './VirtualFocus';
export * from './useHotkeys';
