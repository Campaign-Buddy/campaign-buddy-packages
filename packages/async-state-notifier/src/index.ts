export * from './AsyncStateNotificationProvider';
export * from './ConsoleLoggingAsyncStateNotificationProvider';
export * from './types';
