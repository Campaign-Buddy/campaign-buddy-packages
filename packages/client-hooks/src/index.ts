export * from './fileSystemApi';
export * from './entityApi';
