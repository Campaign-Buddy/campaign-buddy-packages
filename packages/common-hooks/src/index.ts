export * from './useBooleanState';
export * from './useCancelableCallback';
export * from './useDebouncedCallback';
export * from './useDebouncedEffect';
export * from './useIsMounted';
export * from './useStableValue';
export * from './useUpdatingRef';
export * from './useResizeObserver';
export * from './useCombinedRefs';
export * from './useDelayedEffect';
export * from './useDomEventHandler';
export * from './useDomNode';
export * from './useOnClickOutside';
export * from './useTaggedChildren';
