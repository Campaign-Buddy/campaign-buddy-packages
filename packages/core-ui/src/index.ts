export { DefaultTheme } from 'styled-components';
export { defaultTheme } from './theme';

export * from './aggregated-input';
export * from './button';
export * from './form-group';
export * from './icon';
export * from './input';
export * from './layout-primitives';
export * from './list';
export * from './menu';
export * from './modal';
export * from './popover';
export * from './select';
export * from './spinner';
export * from './switch';
export * from './tag';
export * from './text';
export * from './tooltip';
