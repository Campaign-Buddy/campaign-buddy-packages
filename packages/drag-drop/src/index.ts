export * from './PaneDragPreview';
export * from './useSectionedDropZone';
export * from './coordinateTransformers';
export * from './dragDataTypes';
