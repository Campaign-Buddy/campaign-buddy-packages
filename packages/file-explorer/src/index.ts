export * from './FileExplorer';
export * from './EntityFileExplorer';
