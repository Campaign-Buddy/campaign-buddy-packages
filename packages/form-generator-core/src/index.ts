export {
	WidgetLookup,
	WidgetProps,
	UiSectionProps,
	SyncedFormGeneratorProps,
	FormGeneratorProps,
} from './FormGeneratorProps';
export { FormGenerator } from './FormGenerator';
export * from './AggregationSupport';
