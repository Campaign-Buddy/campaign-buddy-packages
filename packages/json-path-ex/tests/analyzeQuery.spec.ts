import { describe, it, expect } from 'vitest';
import { analyzeSubQueries } from '../src';

interface TestCase {
	query: string;
	json: any;
}

const testCases: TestCase[] = [
	{
		query: '$.foo',
		json: {
			foo: 'f',
			baz: 'b',
		},
	},
	{
		query: '$.foo[?(@ > 2)]',
		json: {
			foo: [0, 2, 3, 4],
			baz: 'b',
		},
	},
	{
		query: '$.foo<?(@ === "bar")>',
		json: {
			foo: {
				bar: 'baar',
				bop: 'boop',
			},
			baz: 'b',
		},
	},
	{
		query: '$.foo[?(@.id > 0)]..bings',
		json: {
			foo: [
				{ id: 0, nested: { bings: 'ruhroh' } },
				{ id: 1, nested: { bings: 'okay' } },
				{ id: 2, nested: { bings: 'okay' } },
				{ id: 3, nested: { bings: 'okay' } },
			],
		},
	},
	{
		query: '$.foo..bings',
		json: {
			foo: [
				{ id: 0, nested: { bings: 'ruhroh' } },
				{ id: 1, nested: { bings: 'okay' } },
				{ id: 2, nested: { bings: 'okay' } },
				{ id: 3, nested: { bings: 'okay' } },
			],
		},
	},
	{
		query: '{$.foo.bar} + {$.baz.bop}',
		json: {
			foo: 'bar',
		},
	},
];

describe('analyzeQuery', () => {
	for (const testCase of testCases) {
		it(`query analysis works for query = ${testCase.query}`, () => {
			const result = analyzeSubQueries(testCase.json, testCase.query);
			expect(result).toMatchSnapshot();
		});
	}
});
