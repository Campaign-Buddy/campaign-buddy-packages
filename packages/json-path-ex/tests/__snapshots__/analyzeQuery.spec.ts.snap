// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`analyzeQuery > query analysis works for query = $.foo 1`] = `
{
  "evaluatedExpressions": [],
}
`;

exports[`analyzeQuery > query analysis works for query = $.foo..bings 1`] = `
{
  "evaluatedExpressions": [],
}
`;

exports[`analyzeQuery > query analysis works for query = $.foo<?(@ === "bar")> 1`] = `
{
  "evaluatedExpressions": [],
}
`;

exports[`analyzeQuery > query analysis works for query = $.foo[?(@ > 2)] 1`] = `
{
  "evaluatedExpressions": [],
}
`;

exports[`analyzeQuery > query analysis works for query = $.foo[?(@.id > 0)]..bings 1`] = `
{
  "evaluatedExpressions": [],
}
`;

exports[`analyzeQuery > query analysis works for query = {$.foo.bar} + {$.baz.bop} 1`] = `
{
  "evaluatedExpressions": [
    {
      "expressionKind": "root",
      "path": "$",
    },
    {
      "expressionKind": "childDescent",
      "path": "$",
    },
    {
      "expressionKind": "propertyAccessor",
      "path": "$",
    },
    {
      "expressionKind": "childDescent",
      "path": "$.foo",
    },
    {
      "expressionKind": "propertyAccessor",
      "path": "$.foo",
    },
  ],
}
`;
